### ISSUE TITLE
Multiple Issues observed while running the application

### ISSUE THREAD

**Issue by Searinox** [zlatinb/muwire#66]
Not a standard way to report things, but having used 0.8.9 for some days now and running 2 nodes on 2 different machines, I noticed the following issues, in order of severity:

-Upon sharing all files, even after a few hours, only part of them appear in the search results(tried each machine searching for the files the other shared).

-Files that are found in search results download normally. For those that aren't showing up, Browse host can be used and they can be manually picked and downloaded that way.

-(Continuing from above) If this is done, there is a Hash download after which the download disappears and does not proceed to download the actual file. It may appear as if the download has been canceled since it's not showing up anywhere. However upon restarting MuWire the downloads stuck in limbo are repopulated in the download list and will be listed again. They can then be canceled/removed. The problem here is that the Hash download and the download of the actual File are treated as two separate entries in the Downloads list. If a download gets stuck between them, there is no way to reach and cancel it. I suggest the UI listing contain both steps.

-There is no listing for files/folders that are in the process of being shared. If a folder is shared accidentally, deleting fully shared entries from the Library list will just have the folder pop up again when new files are done being shared. Depending on the size of the shared folder it can take a very long time to wait for it to finish sharing, before being able to remove it for good.

-Created Collections are gone on next MuWire restart.

-Maximum Chat Connections reverts to -1 on MuWire restart.

-It is not possible to select multiple files to download / multiple downloads to cancel. Initiating a download also switches to the Downloads tab so if you're trying to download multiple files, you have to keep switching back to the results tab.

-Once a contact has been added, it is not possible to edit its comment.

**Comment by zlatinb** [comment]
Hello, thanks for the feedback.  There are several issues, I'll try to address each one:

1. Files not appearing in search results but appearing in browse host - this is very serious.  I would ask for your help in debugging this if you can reproduce it.  Please open a separate GitHub issue for this, we'll investigate it there.
2. The download disappears after Hash LIst download - first time I hear about this, but as an uploader I have observed sometimes someone just requests the hash list and does not proceed to download the file, so maybe that's what's happening.  Can you reproduce it?  It's worth a separate GitHub issue.  Unifiying the Uploads table to show hash list uploads and file uploads in the same row should not be difficult, I'll put it on my TODO for the next version.
3. No listing of files/folders waiting to be shared - I understand this is annoying, it would be a big internal change to add such a view.  I'll think about it but can't promise it will be done for the next version
4. Created collections gone - this is also serious, I'll try to reproduce.  Is the MuWire shutdown graceful or do you kill the process?
5. Maximum chat connections reverting - this should be easy to fix, I'll take a look soon.
6. Selecting multiple files to download or multiple downloads to cancel - this shouldn't be difficult to fix.  I'll try to have it fixed for the next version
7. Editing a contact comment - nice to have, will do it if time permits
8. Sizes/positions not remembered - I'll take look, shouldn't be difficult.

Again, thank you for testing MuWire thoroughly, I really appreciate it!

**Comment by Searinox** [comment]
1. Will open an issue.

2. I can reproduce this consistently: I have 2 MuWire installs on 2 machines. Each one can find some files from the other in the results, but not all of them. Tried deleting all shares and re-sharing with 1 thread instead of 2 but it didn't make a difference. The files are kinda random, but once I find one that isn't in the results, I can obtain the the downloads list from the install and browse to download it manually. It doesn't matter which one of the two it is, viceversa works all the same:
-hash list is downloaded
-nothing more is downloaded after, download just disappears, as if it completed or failed(even though I have set infinite retry)
-if MuWire is closed and re-opened the download that disappeared re-appears; it is usually able to complete the download, so I guess whenever I see this happen I know I just have to close and re-open MuWire for the download to proceed properly

I recommend you try to reproduce this the same way: set up 2 nodes and share a few thousand files. Wait for a while until when you'd expect the files to show up in searches. Do some searches, you should find that not all your files on the node are findable. Take note of the ones you aren't finding. Then Browse the entire user's file list. Try to download one of those missing files. What should happen is, you see a download of the "Hash List", then it disappears and never continues. Close and restart MuWire and the download should be back in the list and even successfully complete.

3. Understandable.

4. 100% repro on this one too. I go ahead and create a new collection, put everything I have in it, and it shows in the Collections tab. With the other node I can download the collection and browse it. If I close MuWire and re-open it, the collection I created is no longer there. Happening on both of my installs every single time. I've not had any instance so far of MuWire.exe hanging or needing to be forced closed. It disappears from Task Manager after I shut it down and gracefully also takes the tray icon with it(usually crashing programs leave the tray icon until you mouse over it). Nothing in AppData CrashDumps or Windows' Event Viewer.

5. OK.

6, 7, 8. These were minor reports anyway. No problem.

**Comment by zlatinb** [comment]
Now that 1 and 2 have their own issues, let's look at 4 - disappearing collections.

The way it works is collections get loaded only after all shared files get loaded.  If after the shared files are all loaded the collection you created still isn't there, then that's a problem.  If that's the case please open a separate issue for it.

**Comment by Searinox** [comment]
I went further and investigated. It appears to be mainly a UI issue. More details to follow in the actual issue that I will be opening.

**Comment by zlatinb** [comment]
Hi,

In build 66 I've fixed items 5 (chat connections value) and 6 (multiple selection).  Please test https://muwire.com/downloads/MuWire-0.8.9-GitHub66.zip
